### ISSUE TITLE
Import Facebook JSON archive

### ISSUE THREAD

**Issue by tyteen4a03** [tootsuite/mastodon#14287]
### Pitch

Facebook allows users to download their user / page's data as a JSON archive. Mastodon should support importing them.

### Motivation

I manage prolific pages and accounts on several social networks (including Facebook) and would like to mirror them on Mastodon.

**Comment by ThibG** [comment]
Unless I'm missing something, I don't think this feature would make a lot of sense. Indeed, bulk-importing posts would probably put unreasonable strain on fediverse servers. Furthermore, Facebook and Mastodon have pretty different presentation and expectations, Facebook posts are usually much longer than a Mastodon post, or even longer than what most Mastodon servers allow to be sent.

**Comment by tyteen4a03** [comment]
I'm only interested in importing my own content onto my own instance (with the character limit set to a value that matches Facebook's).

**Comment by ctrlbreak-** [comment]
As someone who is just about fed up with FB and has also exported their data in this manner, I'm completely onboard.  Anything is better than nothing and would only encourage other less technical folks on the fence and considering the jump to Mastodon.

**Comment by john-clark** [comment]
I agree I am leaving facebook and would like to take my historical data to my own platform. I have been evaluating selfhosted distributed social media platforms and being able to import my social media history would make moving to mastadon a no brainer.

**Comment by lovebes** [comment]
It's either this or just pushing into wordpress or some static website. It would be amazing if Mastodon would accomplish this.

**Comment by TheBestPessimist** [comment]
+1 for the Facebook import. 

I'm fine with this bulk import having a very low priority on the server side, even lasting for a whole hour (the user being informed about it ofc), as long as it's still available.

Regarding the past length issue: I believe that is a non issue: split each long post in multiple comments, just like we often see in Twitter with 
"1/many"
"2/many"
etc

**Comment by harrybailey** [comment]
This issue is likely missing the 'suggestion' label

**Comment by robusticus** [comment]
It seems like this can be done but the only issue is the posted or created date. If you can't set that, and everything is posted on the same day you lose a lot.

**Comment by Sieboldianus** [comment]
> It seems like this can be done but the only issue is the posted or created date. If you can't set that, and everything is posted on the same day you lose a lot.

I think it would be fine to only allow this for administrators (e.g. for owned instances). In this case, modification of `post_create_date` in the database (postgres) would be possible.

**Comment by robusticus** [comment]
> I think it would be fine to only allow this for administrators (e.g. for owned instances). In this case, modification of `post_create_date` in the database (postgres) would be possible.

Would you mind explaining why you think this ought to be locked down to admins? Ideally this would be an option when you sign up for Mastodon. MVP phase 1 though is looking like a Bash/Powershell script and a PR for a new import_status method on the API that will override create_at. Having to do this on my own instance won't deter me but I have absolutely no desire for my own instance, nor any sort of reasonable plan to fund it.

**Comment by Sieboldianus** [comment]
Apologies - it was not meant to exclude users. Initially, I thought allowing anyone to override `create_date` would allow abuse (e.g. deleting content, importing again, changing publication dates etc.) - but thinking about it, there's no reason why it should be restricted, apart from performance benefits running this on the server in the shell, by an administrator, instead of over the web by any user. Administrators usually know better what the hardware can handle. Anyway, take what I said with a grain of salt, generally I am against exclusive features, so I agree to your point.

**Comment by ThosHats** [comment]
I agree and strongly vote for this feature. Many people have used Facebook as diary for the last 15 years. It would be sad to loose this timeline.

**Comment by filmo** [comment]
@ThosHats as a FB user contemplating the move, I also support importing my FB history, at least in some limited form, particularly if I'm hosing my own federated server. Ideally, on my own server, I could allow my 'circle' to do so as well, to limits defined by me as the admin. 

IMHO, the lesson we (or at least me) learned was that it's the 'history' as well as the connections that make FB so 'sticky' and hard to leave. Who really wants to lose photos and post from 10+ years of interaction?

**Comment by kiang** [comment]
+1 for this. It would be easier to promote Mastodon if people could migrate historical data from Facebook.
